fork-off/src/main.rs

```diff
@@ -0,0 +1,132 @@
+use clap::Parser;
+use serde_json::Value;
+use std::fs::{self, File};
+use std::io::{BufReader, ErrorKind, Read, Write};
+use substrate_api_client::extrinsic::log::info;
+
+#[derive(Debug, Parser)]
+#[clap(version = "1.0")]
+pub struct Config {
+    /// URL address of the node RPC endpoint for the chain you are forking
+    #[clap(long, default_value = "http://127.0.0.1:9933")]
+    pub http_rpc_endpoint: String,
+
+    /// path to write the initial chainspec of the fork
+    /// as generated with the `bootstrap-chain` command
+    #[clap(long, default_value = "../docker/data/chainspec.json")]
+    pub fork_spec_path: String,
+
+    /// where to write the forked genesis chainspec
+    #[clap(long, default_value = "../docker/data/chainspec.fork.json")]
+    pub write_to_path: String,
+
+    /// which modules to set in forked spec
+    #[clap(long, default_value = "Aura, Aleph")]
+    pub prefixes: Vec<String>,
+}
+
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+    let Config {
+        http_rpc_endpoint,
+        fork_spec_path,
+        write_to_path,
+        prefixes,
+    } = Config::parse();
+
+    env_logger::init();
+
+    info!(
+        "Running with config: \n\thttp_rpc_endpoint {}\n \tfork_spec_path: {}\n \twrite_to_path{}",
+        http_rpc_endpoint, fork_spec_path, write_to_path
+    );
+
+    let mut fork_spec: Value = serde_json::from_str(&read_file(&fork_spec_path))?;
+
+    // get current chain state (storage)
+    let storage: Value = reqwest::Client::new()
+        .post(http_rpc_endpoint)
+        .json(&serde_json::json!({
+            "jsonrpc": "2.0",
+            "id": 1,
+            "method": "state_getPairs",
+            "params": ["0x"]
+        }))
+        .send()
+        .await
+        .expect("Storage request has failed")
+        .json()
+        .await
+        .expect("Could not deserialize response as JSON");
+
+    let storage = storage["result"].as_array().expect("No result in response");
+
+    info!("Succesfully retrieved chain state");
+
+    // move the desired storage values from the snapshot of the chain to the forked chain genesis spec
+    info!(
+        "Looking for the following storage items to be moved to the fork: {:?}",
+        prefixes
+    );
+
+    storage
+        .iter()
+        .filter(|pair| {
+            prefixes.iter().any(|prefix| {
+                let pair = pair.as_array().unwrap();
+                let storage_key = pair[0].as_str().unwrap();
+                storage_key.starts_with(&format!("0x{}", prefix_as_hex(prefix)))
+                    || storage_key.eq("0x3a636f6465") // code
```

Comment:
optional: I would firstly map pallets to hex strings, append there `code` and then use it here

Reply:
good tip! Done.